include/mitsuba/core/distr_1d_cond.h

```diff
@@ -0,0 +1,1225 @@
+     * \return
+     *  The value of the normalized PDF at position \c pos, conditioned
+     * on \c cond
+     */
+    Value eval_pdf_normalized(Value pos, std::vector<Value> &cond,
+                              Mask active = true) const {
+        if (cond.size() != m_nodes_cond.size())
+            Log(Error, "The number of conditionals should be %u instead of %u",
+                m_nodes_cond.size(), cond.size());
+
+        if (m_enable_sampling) {
+            auto [value, integral] = lookup(pos, cond, 0, 0, active);
+            return value * dr::rcp(integral);
+        } else {
+            Throw(
+                "You should enable sampling if the normalized pdf is required");
+        }
+    }
+
+    /**
+     * \brief Sample the distribution given a uniform sample \c u, conditioned
+     * on \c cond
+     *
+     * \param u
+     *  Uniform sample
+     * \param cond
+     *  Conditionals where the PDF is sampled
+     * \param active
+     *  Mask of active lanes
+     * \return
+     *  A pair where the first element is the sampled position and the
+     * second element the value of the normalized PDF at that position
+     * conditioned on \c cond
+     */
+    std::pair<Value, Value> sample_pdf(Value u, std::vector<Value> &cond,
+                                       Mask active) const {
+        if (cond.size() != m_nodes_cond.size())
+            Log(Error, "The number of conditionals should be %u instead of %u",
+                m_nodes_cond.size(), cond.size());
+
+        if (m_nearest)
+            Throw("Sampling with nearest neighbor not supported");
+
+        if (m_enable_sampling) {
+            std::vector<Index> indices(1 << m_nodes_cond.size());
+            std::vector<Value> weights(1 << m_nodes_cond.size());
+
+            lookup_fill(cond, 0, 1.f, indices, weights, 0, 0, active);
+            return lookup_sample(u, indices, weights, true, active);
+        } else {
+            Throw("For sampling this distribution, first enable it in the "
+                  "constructor.");
+        }
+    }
+
+    /// Is the distribution object empty/uninitialized?
+    bool empty() const { return m_pdf.empty(); }
+
+    /// Return the maximum value of the distribution
+    ScalarFloat max() const { return m_max; }
+
+    /**
+     * \brief Return the integral of the distribution conditioned on \c cond
+     *
+     * \param cond
+     *   Conditionals that define the distribution
+     * \return
+     *   The integral of the distribution
+     */
+    Value integral(std::vector<Value> &cond) const {
+        if (m_enable_sampling) {
+            const Value dummy_pos  = dr::gather<Value>(m_nodes, Index(0));
+            auto [value, integral] = lookup(dummy_pos, cond, 0, 0, true);
+            return integral;
+        } else {
+            Throw("You should enable sampling if the integral is needed");
+        }
+    }
+
+private:
+    std::pair<Value, Value> lookup(Value pos, std::vector<Value> &cond_,
```

Comment:
I'm under the impression that this function would be more readable if we handled the case `length == 1` in a separate `if` at the top rather than scattered throughout the function.